src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java

```diff
@@ -329,13 +324,60 @@ private void addInvalidDisposalDateError(List<HearingTypeItem> hearingTypeItems,
             return;
         }
 
-        Optional<DateListedTypeItem> hearingTypeItem = findHearingTypeItem(hearingTypeItems, disposalDate);
-
-        if (hearingTypeItem.isEmpty() || !YES.equals(hearingTypeItem.get().getValue().getHearingCaseDisposed())) {
+        if (!disposedHearingWithSameDisposalDateOfJurisdictionExists(hearingTypeItems, disposalDate)) {
             errors.add(String.format(DISPOSAL_DATE_HEARING_DATE_MATCH, jurCode));
         }
     }
 
+    /**
+     * Checks if hearing type items has any hearing type which is disposed
+     * and has the same disposal date with the jurisdiction disposal date.
+     * @param hearingTypeItems All hearings
+     * @param disposalDate Disposal date of Jurisdiction
+     * @return boolean  if true then hearing list has disposed hearing with the
+     *                  same disposal date of jurisdiction.
+     */
+    private boolean disposedHearingWithSameDisposalDateOfJurisdictionExists(
+            List<HearingTypeItem> hearingTypeItems, String disposalDate) {
+        if (CollectionUtils.isEmpty(hearingTypeItems)) {
+            return false;
+        }
+        for (HearingTypeItem hearingTypeItem : hearingTypeItems) {
+            HearingType hearingType = hearingTypeItem.getValue();
+            if (ObjectUtils.isEmpty(hearingType) || CollectionUtils.isEmpty(hearingType.getHearingDateCollection())) {
+                continue;
+            }
+            if (hasDisposedHearingWithSameDisposalDateOfJurisdiction(
+                    hearingType.getHearingDateCollection(), disposalDate)) {
```

Comment:
Sorry to be pedantic but does this fit on a single line now?

Reply:
It fits Harpreet :). Already implemented it to fit on one column but I think it is showing like this on git :)